packages/react-instantsearch-core/src/lib/voiceSearchHelper/index.ts

```diff
@@ -0,0 +1,131 @@
+const STATUS_INITIAL = 'initial';
+const STATUS_ASKING_PERMISSION = 'askingPermission';
+const STATUS_WAITING = 'waiting';
+const STATUS_RECOGNIZING = 'recognizing';
+const STATUS_FINISHED = 'finished';
+const STATUS_ERROR = 'error';
+
+type QueryChangeListener = (query: string) => void;
+type StateChangeListener = () => void;
+
+export type VoiceListeningState = {
+  status: string;
+  transcript?: string;
+  isSpeechFinal?: boolean;
+  errorCode?: string;
+};
+
+export type ToggleListening = () => void;
+
+export default function voiceSearchHelper() {
+  const SpeechRecognitionAPI: new () => SpeechRecognition =
+    (window as any).webkitSpeechRecognition ||
+    (window as any).SpeechRecognition;
+  const getDefaultState = (status: string): VoiceListeningState => ({
+    status,
+    transcript: undefined,
+    isSpeechFinal: undefined,
+    errorCode: undefined,
+  });
+  let state: VoiceListeningState = getDefaultState(STATUS_INITIAL);
+  let recognition: SpeechRecognition | undefined;
+  let searchAsYouSpeak;
```

Comment:
Should this have a default value?

Reply:
It is meant to be set by the connector, but I will put a default value to keep consistency because I put default values for onQueryChange and onStateChange.